core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -914,6 +916,7 @@ protected final void writerSegmentStats(SegmentsStats stats) {
 
     @Override
     public long getIndexBufferRAMBytesUsed() {
+        // We don't guard w/ readLock here, so we could throw AlreadyClosedException
```

Comment:
Well, where `IndexShard` calls this method, it expects/handles the `AlreadyClosedException` (`handleRefreshException` in `IndexShard.java`), and an ACE thrown from here can happen under normal usage (does not necessarily mean there's a bug).

`super`'s javadocs also state that it can throw `AlreadyClosedException`.

We could alternatively acquire the `readLock`, in this method (and re-throw to `AssertionError`), but that's somewhat scary since `writeLock` can be held for quite some time, blocking `IndexingMemoryController` from polling the shards when one engine is flushing.